components/sidebar/sidebar_channel/sidebar_direct_channel/sidebar_direct_channel.tsx

```diff
@@ -0,0 +1,143 @@
+    teammate?: UserProfile;
+    currentTeamName: string;
+    currentUserId: string;
+    redirectChannel: string;
+    active: boolean;
+    botIconUrl: string | null;
+    actions: {
+        savePreferences: (userId: string, preferences: PreferenceType[]) => Promise<{data: boolean}>;
+    };
+};
+
+type State = {
+    svgErrorUrl: string | null;
+};
+
+class SidebarDirectChannel extends React.PureComponent<Props, State> {
+    handleLeaveChannel = (callback: () => void) => {
+        const id = this.props.channel.teammate_id;
+        const category = Constants.Preferences.CATEGORY_DIRECT_CHANNEL_SHOW;
+
+        const currentUserId = this.props.currentUserId;
+        this.props.actions.savePreferences(currentUserId, [{user_id: currentUserId, category, name: id!, value: 'false'}]).then(callback);
+
+        trackEvent('ui', 'ui_direct_channel_x_button_clicked');
+
+        if (this.props.active) {
+            browserHistory.push(`/${this.props.currentTeamName}/channels/${this.props.redirectChannel}`);
+        }
+    }
+
+    onSvgLoadError = () => {
+        this.setState({
+            svgErrorUrl: this.props.botIconUrl,
+        });
+    }
+
+    onSvgLoad = () => {
+        this.setState({
+            svgErrorUrl: null,
+        });
+    }
+
+    getIcon = () => {
+        const {channel, teammate} = this.props;
+
+        if (!teammate) {
+            return null;
+        }
+
+        if (teammate.id && teammate.delete_at) {
+            return (
+                <i className='icon icon-archive-outline'/>
+            );
+        } else if (teammate.id && teammate.is_bot) {
+            // Use default bot icon
+            let icon = (<i className='icon icon-robot-happy'/>);
+
+            // Attempt to display custom icon if botIconUrl has changed
+            // or if there was no error when loading custom svg
+            if (this.props.botIconUrl &&
+                this.props.botIconUrl !== this.state.svgErrorUrl) {
+                icon = (
+                    <Svg
+                        className='icon icon-robot-happy'
+                        src={this.props.botIconUrl}
+                        onLoad={this.onSvgLoad}
+                        onError={this.onSvgLoadError}
+                    />
+                );
+            }
+            return icon;
+        }
+
+        let className = '';
+        if (channel.status === 'online') {
+            className = 'status-online';
+        } else if (channel.status === 'away') {
+            className = 'status-away';
+        } else if (channel.status === 'dnd') {
+            className = 'status-dnd';
+        }
+
+        return (
+            <StatusIconNew
+                status={channel.status}
+                className={className}
+            />
+        );
+    }
+
+    render() {
+        const {channel, teammate, currentTeamName} = this.props;
+
+        if (!teammate) {
+            return null;
+        }
+
+        let displayName = channel.display_name;
```

Comment:
Where are these extra props on the channel (namely `display_name` for DMs and `status`) coming from? The code that we're using now shouldn't be adding them to the channel object as far as I know

Reply:
They are part of the `Channel` object type, I haven't added them in anywhere.

Reply:
The `display_name` should exist on the channel, but some of the other fields (`isCurrent`, `teammate_id`, `status`) are injected by some of the selectors as well as the `display_name` which is overridden for DMs/GMs. I don't think any of my selectors add those though, so I don't know where they'd be coming from.

Reply:
Oh, they're being injected by [`makeGetChannel`](https://github.com/mattermost/mattermost-redux/blob/master/src/selectors/entities/channels.ts#L112). I'm going to file a ticket to move away from that selector, but we can do that later.